doc/adrs/XXX_support-server-side-pagination-for-table.adoc

```diff
@@ -0,0 +1,108 @@
+= ADR-XXX Support Server-Side Pagination for Table Representation
+
+== Context
+
+The default pagination features of Material React Table are client-side.
+This means you have to have all of your data fetched and stored in the table all at once.
+This may not be ideal for large datasets.
+
+== Decision
+
+To address the limitations of client-side pagination, we will implement server-side pagination for Material React Table.
+This approach will allow us to fetch and display data more efficiently, especially for large datasets.
+The server-side pagination method will only be applied to table representations.
+For the use of tables as form widgets, client-side pagination will continue to be used.
+
+=== Material React Table
+
+To enable server-side pagination, we will set `manualPagination: true` in Material React Table.
+This will allow us to take control of pagination on the backend, ensuring that we can efficiently manage large datasets and accurately display the true total number of rows.
+The pagination must be managed within our own state, and listeners must be added to handle pagination state changes, allowing the backend to be called with the updated pagination data.
+
+[source, typescript]
+----
+const [pagination, setPagination] = useState({
+  pageIndex,
+  pageSize,
+});
+
+useEffect(() => {
+  //do something when the pagination state changes
+}, [pagination.pageIndex, pagination.pageSize]);
+
+const table = useMaterialReactTable({
+  onPaginationChange: setPagination,
+  state: { pagination },
+});
+
+
+----
+
+=== Frontend
+
+The pagination data will be added to the `representationId` of the subscription:
+
+[source, typescript]
+----
+`${tableId}?pageIndex=${pageIndex}&pageSize=${pageSize}`
```

Comment:
I agree but we need to keep in mind that we will end up with multiple versions of the table in memory and then when the end users will want to perform some synchronized operation like activating some filter for all users we would have to consider this input (or the change description) in multiple representations. This will have an impact in quite a unique way for future development.

We never had to handle something like this in trees for example because changes are either semantic or local to one instance.

Reply:
Yes, I think the pagination is the only data that we don't want to be synchronized between all the open representation.

Reply:
It's doable to handle all the other concerns but the synchronization of the filtering state for example across all instances of a specific table with various instances depending on the pagination state of each subscription will be tricky. We will need to talk about this in a future dedicated ADR. We should keep that in mind as a consequence of this ADR.